Common room

gunmack opened the pull request:
- Photo of ASB 9802 + description of the room + directions (maybe link to SFU map?)
          - Added placeholder image
          - Added clickable link to sfu room finder
          - TODO: description

- Photo of ASB 9971 + description of the room + directions (maybe link to SFU map?)
         - same as 9802 

- Information about our vending machines + items (typically) sold + the costs
          - Added two types of tables for the 2 different machines ( lmk which one looks better )
          - TODO: Maybe put a photo of a vending machine?

- Footer (use the component provided in /src/main/components/Footer.js - for reference, see how its used in /src/main/pages/Home.js ( DONE )

![image](https://github.com/user-attachments/assets/9fba41d1-6339-49dd-81ed-a65d9f685630)
![image_2](https://github.com/user-attachments/assets/89059ace-80ba-4ec3-9829-9262e925c685)

EarthenSky commented on src/main/pages/CommonRooms.js:
```diff
@@ -0,0 +1,216 @@
+                                        {/* <th>Popularity</th> */}
+                                    </tr>
+                                </thead>
+                            {pop.map((pop,index) => (
+                                <tbody>
+                                    <tr>
+                                        <td className='text-black text-center px-4 bg-slate-400'>{pop.name}</td>
+                                        <td className='text-cyan-200 text-center bg-black'>{pop.price}</td>
+                                        {/* <td>Number of stars</td> */}
+                                    </tr>
+                                </tbody>
+                            ))}
```
The colours for the tables are a bit intense (the blue on red is the most painful). I encourage you to look up [some references](https://www.google.com/search?sca_esv=ff89fd480319a86b&sca_upv=1&sxsrf=ADLYWILnYK0vDc6nIkUrKJYpL6Xu32xtSA:1723934321896&q=html+tables+UI+design&udm=2&fbs=AEQNm0DYVld7NGDZ8Pi819Yg8r6em07j6rW9d2jUMtr8MB7htoxbI0iAKNRPykigVf3e9aputkbr8jzmN5LYbANOqrq5HYnx4MjtyMxZ94LvgeHWmGBcuWUoydKfNaoB5JMdZlMtXmg2De2y5O7nn-eTbNdYHsRiT1RQ-pB6qp3ejXJ5VpdCk5NA1Jug5hVR16L7F-A1C1p-4xpfp7qj2HsGNaipPZQOiw&sa=X&ved=2ahUKEwjrr_3mi_2HAxUtJjQIHaHZMwIQtKgLegQIExAB&biw=1745&bih=859&dpr=1.1#vhid=JPIKcw5VuiOb9M&vssid=mosaic) & [take inspiration](https://dev.to/dcodeyt/creating-beautiful-html-tables-with-css-428l) from them

micahdbak replied:
Agree with Gabe a bit here; the red is fine, but you could replace the cyan colours with X-gray-YYY; see https://tailwindcss.com/docs/customizing-colors

EarthenSky commented on src/main/pages/CommonRooms.js:
```diff
@@ -0,0 +1,216 @@
+                            such as ⤵️
+                    </p>
+                           
+                    <ul className="list-disc list-inside text-lg pl-8" style={{ fontFamily: 'Poppins, sans-serif' }}>
+                        <li> Fridge 🧊</li>
+                        <li> Microwave 🔥</li>
+                        <li> Free Coffee ☕</li>
+                        <li> Free milk and cereal 🥛</li>
+                        <li> Board games ♠️</li>
+                        <li> A place to hang out with friends and meet new people 😎</li>
+                    </ul>
```
Can we put the emoji on the left? I think it might look better if they're all aligned. Just an idea!

That being said, this page has a bit too many emoji...

gunmack replied:
Noted, it seemed a little plain so I added some emojis but got carried away lol

EarthenSky replied:
The list might actually work nicely, but the snacks & drinks is a bit repetitive. Not saying you should remove everything, but maybe we can consider some other ways to add colour?

One possible way is to add some little textured banners [like red blob games](https://www.redblobgames.com/x/1843-planet-generation/) does. The following are two little examples I made with our black/grey colour scheme, though if you want nicer complementary colours you could look for a nice soft [palette here](https://lospec.com/palette-list)

![crossed-tiled-banner](https://github.com/user-attachments/assets/86912f29-01dc-4606-93ea-2f3a70a4f177)
![diamond-texture](https://github.com/user-attachments/assets/4b0f0a15-4100-4a9d-8c5c-3598e1b5e8c5)

micahdbak replied:
@gunmack , I personally disagree with Gabe on the emojis (I like them), so I say it's up to you. Don't feel pressured to keep or remove them, do what you think looks good

Same with the background banners

EarthenSky reviewed (changes requested):
This is awesome! I added a few design considerations

One final design consideration is the alignment of the titles for the table: they should have the same y position, and it’s okay if the bottoms of the tables don’t line up.

In addition, let's include the following:
- a [roomfinder link](https://roomfinder.sfu.ca/apps/sfuroomfinder_web/) to the CR
- an svg map of the CR - I'll convert [this pdf](https://www.sfu.ca/content/sfu/fs/campus-maps/key-plans/burnaby-campus/_jcr_content/main_content/download_1217477394/file.res/Applied%20Science%20Building%209000%20Level.pdf) to svg & send it to you by EOD ([sfu has maps here](https://www.sfu.ca/fs/campus-maps/key-plans/burnaby-campus.html))

gunmack commented:
Noted, will start on changes tomorrow.

micahdbak commented on README.md:
```diff
@@ -29,7 +29,7 @@ Run `npm run start` to deploy the site locally at [localhost:8080](http://localh
 This script starts a Webpack development server, and watches any changes made to the `src` and `public` directories.
 
-If you start using new Tailwind classes, however, make sure to run `npm run tailwind` to refresh the `public/static/css/tailwind.css` file.
+If you start using new Tailwind classes, however, make sure to run `npm run tailwind` or `npm run tw` to refresh the `public/static/css/tailwind.css` file.
```
Oh my, good catch!

The former doesn't work I think, so if you could replace this with just `npm run tw` instead of `npm run tailwind` that would be great.

Thanks!

micahdbak commented on src/main/pages/CommonRooms.js:
```diff
@@ -0,0 +1,216 @@
+                </div>
+
+                {/*Photos of the common rooms*/}
+                <div className='mx-16 my-8 flex gap-8 '>
```
Could you make this a flex column on smaller displays with

```js
<div className='mx-16 my-8 flex flex-col md:flex-row gap-8'>
```

? A flex row gets a little squashed for mobile screens!

micahdbak commented on src/main/pages/CommonRooms.js:
```diff
@@ -0,0 +1,216 @@
+
+                {/*Photos of the common rooms*/}
+                <div className='mx-16 my-8 flex gap-8 '>
+                    <div>
```
As these two sections seem to be the same, and they're using data from the array `rooms`, you could do some cool code dynamics like so:

```js
            {rooms.map(room => {
                return (
                    <div>
                        <a href={room.link}>
                            <div className="relative w-full h-full overflow-hidden rounded-2xl transform transition-transform duration-300 hover:scale-125">
                                <img
                                    src={room.image}
                                    alt={room.text}
                                    className="w-full h-full "
                                />
                                <div className="absolute inset-0 bg-gradient-to-r from-black to-transparent opacity-90"></div>
                                <div className="absolute inset-y-0 left-0 flex items-center pl-8">
                                    <h2
                                        className="text-xl sm:text-2xl md:text-3xl font-bold text-white"
                                        style={{ fontFamily: 'Poppins, sans-serif' }}
                                    >
                                        {rooms[0].text}
                                    </h2>
                                </div>
                            </div>
                        </a>
                    </div>
                );
            })}
```

micahdbak replied:
Oh also, I'd recommend storing the room finder links for these clickables; if you have a different plan for them that's ok too!

micahdbak reviewed (commented):
I'm loving the feel of this so far! I think Gabe was a bit turned off but I'm a fan, so feel free to keep it as is or aim for somewhere inbetween, or where Gabe wants - up to you.

You'll want to pull in new changes from main as I just merged two PRs; thanks so much for the progress you've made!!

I'll also get back to you with some photos for either common rooms

micahdbak commented:
@gunmack - poke poike

peyz21 commented:
Ship it 🗣️

gunmack commented:
![image](https://github.com/user-attachments/assets/c3fd0e37-ce22-471e-abb1-b845b217e1b6)
![image](https://github.com/user-attachments/assets/037e4d57-e717-47ca-82be-8a0472856785)

- Lined up the emojis
- Added room finder links to the clickable images
- decided to keep the tables simple and just use the two colours that are already present throughout
- renamed header from `Snacks and drinks` to `Refreshments`

### TODO: 

- Interior images of the common rooms

- Maybe some more words describing the rooms, if required

- @EarthenSky, I am not sure how I should use the svg files here. 
       - Clickable links that redirect to the svg or
       - Embed as images on the page 
       
- One last thing to add is `Popularity` column for each item 
            - need to ask Aiya about that unless someone else can give me some data for it somehow

micahdbak commented on src/main/pages/Home.js:
```diff
@@ -80,7 +80,7 @@ export const Home = () => {
             Welcome to the
             <br />
-            Computing Science Student Society
+            Computing Science Student Society 
```
Blank space here!

micahdbak replied:
At the end of the line

micahdbak commented:
I'm happy to approve this PR given the following two things:

- Run `npm run lint` as a final commit (will lint all JS files as per specifications in `.prettierrc`)
- Real CR images are used (ask Jake for these)

EarthenSky commented on public/static/files/new-common-room.svg:
actually, using them as images stop the text in them from being selectable; it's better to use them as an object like `<object height="500" data="/static/files/new-common-room.svg"></object>`

EarthenSky commented:
>  I am not sure how I should use the svg files here

I was thinking we'd place them above / below the clickable link buttons, so that the page would have a map of the CR, but if you can't think of a good place for them, we can always add them later. No worries!

gunmack commented:
- Added the svg files, had to do a white background cause all the text on 9971 map mixes into the background.
![image](https://github.com/user-attachments/assets/e146698d-4c72-4c72-a2dd-5c07e2f75c61)
- Just need to source a picture of 9802 and that should be all.

gunmack commented:
![image](https://github.com/user-attachments/assets/96de77cc-ae7b-4c9e-b4a2-6e86c7f14403)

gunmack commented on public/static/files/main/homepage.jpg:
idk how this got into my commit, lmk if I need to `git rm` it and push again

EarthenSky replied:
it was when you merged main into this branch; we replaced it with a smaller image (12MB way too big for web!). Definitely `git rm` pls!

EarthenSky replied:
actually, I got you though

EarthenSky commented:
Re: https://github.com/CSSS/csss-site-frontend/pull/42/files (small aliasing artifacts that look to be a bug in tailwind iirc)

**before**

https://github.com/user-attachments/assets/82bbd968-b95b-4aa6-bccd-b5003edb3713

**after**

https://github.com/user-attachments/assets/ac0a64b3-5cb3-4714-be47-55ed609a191e

gunmack commented:
awesome, thanks @EarthenSky !!!
